### ISSUE TITLE
Unfortunately, AppName has stopped.

### ISSUE THREAD

**Issue by avradu1984** [myflashlab/GameServices-ANE#17]
I am trying to integrate several ANEs into my game and I am having problems with Game Services.

I get "Unfortunately, AppName has stopped" after GameServices.google.auth.login(); is called: the login overlay appears but, when it closes, that message appears and it crashes.

Additionally, that message and crash stop happening if I include firebaseCore.ane in the project. The crash no longer happens, but I get a new message: "AppName is having trouble with Google Play services. Please try again."

This is my manifest:

```xml
<application xmlns="http://ns.adobe.com/air/application/25.0">
        <id>wms.wordify</id>
        <versionNumber>1.0.0</versionNumber>
        <versionLabel/>
        <filename>Game Services</filename>
        <description/>
        <name>Game Services</name>
        <copyright/>
        <initialWindow>
                <content>gameservices.swf</content>
                <systemChrome>standard</systemChrome>
                <transparent>false</transparent>
                <visible>true</visible>
                <fullScreen>true</fullScreen>
                <aspectRatio>portrait</aspectRatio>
                <renderMode>gpu</renderMode>
                <autoOrients>false</autoOrients>
        </initialWindow>
        <icon/>
        <customUpdateUI>false</customUpdateUI>
        <allowBrowserInvocation>false</allowBrowserInvocation>
        <android>
                <manifestAdditions>
                        <![CDATA[<manifest android:installLocation="preferExternal">
                        <uses-sdk android:minSdkVersion="16" android:targetSdkVersion="22"/>
                        <uses-permission android:name="android.permission.INTERNET"/>
                        <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>
                        <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE"/>
                        <uses-permission android:name="android.permission.ACCESS_WIFI_STATE"/>
                        <uses-permission android:name="com.android.vending.BILLING" />
                        
                        <application>
                        
                        <activity>
                        <intent-filter>
                        <action android:name="android.intent.action.MAIN" />
                        <category android:name="android.intent.category.LAUNCHER" />
                        </intent-filter>
                        <intent-filter>
                        <action android:name="android.intent.action.VIEW" />
                        <category android:name="android.intent.category.BROWSABLE" />
                        <category android:name="android.intent.category.DEFAULT" />
                        </intent-filter>
                        </activity>
                        
                        <!-- application ID which identifies your game settings in the Google Game Services console (NOTE: leave the initial '\ ' as it is) -->
                        <meta-data android:name="com.google.android.gms.games.APP_ID" android:value="\ 909174111923" />
                        <meta-data android:name="com.google.android.gms.version" android:value="@integer/google_play_services_version" />
                        
                        </application>
                        </manifest>]]>
                </manifestAdditions>
        </android>
        
  
  
  
  <extensions>
    <extensionID>com.myflashlab.air.extensions.dependency.androidSupport</extensionID>
    <extensionID>com.myflashlab.air.extensions.dependency.googlePlayServices.appinvite</extensionID>
    <extensionID>com.myflashlab.air.extensions.dependency.googlePlayServices.auth.base</extensionID>
    <extensionID>com.myflashlab.air.extensions.dependency.googlePlayServices.base</extensionID>
    <extensionID>com.myflashlab.air.extensions.dependency.googlePlayServices.basement</extensionID>
    <extensionID>com.myflashlab.air.extensions.dependency.googlePlayServices.drive</extensionID>
    <extensionID>com.myflashlab.air.extensions.dependency.googlePlayServices.games</extensionID>
    <extensionID>com.myflashlab.air.extensions.dependency.googlePlayServices.plus</extensionID>
    <extensionID>com.myflashlab.air.extensions.dependency.googlePlayServices.tasks</extensionID>
    <extensionID>com.myflashlab.air.extensions.dependency.overrideAir</extensionID>
    <extensionID>com.myflashlab.air.extensions.gameServices</extensionID>
    <extensionID>com.myflashlab.air.extensions.dependency.firebase.analytics.impl</extensionID>
    <extensionID>com.myflashlab.air.extensions.dependency.firebase.common</extensionID>
    <extensionID>com.myflashlab.air.extensions.dependency.firebase.iid</extensionID>
    <extensionID>com.myflashlab.air.extensions.dependency.firebase.analytics</extensionID>
    <extensionID>com.myflashlab.air.extensions.firebase.analytics</extensionID>
    <extensionID>com.myflashlab.air.extensions.firebase.core</extensionID>
  </extensions>
</application>

```

And this is my code:

```actionscript
package com.avr {
	import com.myflashlab.air.extensions.gameServices.GameServices;

	// All events are inside the com.myflashlab.air.extensions.gameServices.google.events package
	import com.myflashlab.air.extensions.gameServices.google.events.AuthEvents;

	// on a successful login, you will receive some information about the signed-in user (NO, you will not receive email address or access token of the user here!)
	import com.myflashlab.air.extensions.gameServices.google.player.Player;
	import com.myflashlab.air.extensions.gameServices.google.player.PlayerLevel;


	public class GoogleGameServices {

		public function GoogleGameServices() {
			init()
		}
		public function init(): void {
			// initialize the Game Services and wait for a successful init call before doing anything else
			try {
				GameServices.init(true); // set this to 'true' when you are building for production (or adHoc on iOS)
				GameServices.google.auth.addEventListener(AuthEvents.INIT, onInit);
				GameServices.google.auth.addEventListener(AuthEvents.LOGIN, onLoginSuccess);
				GameServices.google.auth.addEventListener(AuthEvents.ERROR, onLoginError);
			} catch (e: Error) {
				
			}
		}
		private function onInit(e: AuthEvents): void {
			// status code indicating whether Game Services can run in your app or not.
			if (e.status == GameServices.SUCCESS) {
				GameServices.google.auth.login();
			} else {}
			trace("onInit result = " + e.status);
		}
		private function onLoginSuccess(e: AuthEvents): void {
			trace("player signed in – ready to use google games services!");
		}


		private function onLoginError(e: AuthEvents): void {
			trace("something went wrong signing in.");
		}
		private function onSignedOut(): void {
			trace("player signed out.");
		}

	}

}

```

Any help will be highly appreciated!

Thanks!

**Comment by myflashlab** [comment]
Please use the ANE alone first because like this will add confusion. For example, I do not see the Firebase settings in the manifest. Anyhow, you said you tested alone and you saw the crash there. so, let's stick to that and try to solve the problem.

The first thing I noticed is that you are using AIR SDK 25 but you have set the ```android:targetSdkVersion="22"```. I'm not sure this is safe. considering that you have AIR 25, it's a good idea to use at least androidtargetSdk 23.

The other important thing is to always check for silent login ability:
```actionscript
GameServices.google.auth.addEventListener(AuthEvents.TRYING_SILENT_LOGIN, onSilentTry);

private function onSilentTry(e:AuthEvents):void
{
	trace("canDoSilentLogin = ", e.canDoSilentLogin);
	trace("If silent try is \"false\", it means that you have to login the user yourself using the \"GameServices.login();\" method.");
	trace("But if it's \"true\", it means that the user had signed in before and he will be signed in again silently shortly. and you have to wait for the \"AuthEvents.LOGIN\" event.");
	
	if (e.canDoSilentLogin)
	{
		trace("connecting to Game Services, please wait...");
	}
	else
	{
		// You should always let users click the "login" button themselves! This is just a sample though and we are doing it programmatically
		GameServices.google.auth.login();
	}
}
```
Finally, make sure you are using the the latest versions of the ANEs and dependencies.

If you still have the problem, please post the AndroidStudio logcat so we can also see the crash logs and can better assist you.

**Comment by myflashlab** [comment]
Just released V4.0.0 of this ANE and this ANE now uses the [GoogleSignIn ANE](https://github.com/myflashlab/GoogleSignIn-ANE) to take care of the authentication. Update yourself and make sure to read the new Wiki to see how you should do things. If any problem, please open a new issue because authentication inside the GameServices ANE is no longer relevant.
